File: submissions/lumen/06/python/1/pizza.py
def f(c):
    x=2000
    y=5000
    z=500
    w=100
    if c<x and c%w==0:
        b=c
    elif c<x:
        b=(int(c/w)+1)*w
    elif c<y and c%z==0:
        b=c
    elif c<y:
        b=(int(c/z)+1)*z
    else:
        s=int(c*1.1)
        d = s%z
        if d<250:
            b=s-d
        else:
            b=s-d+z
    t=b-c
    if t==0 or t>1500:
        print("PAID BY CARD")
    elif t%w!=0:
        print("KEEP THE CHANGE YOU FILTY ANIMAL")
    else:
        print("HERE IS {0}".format(b))
    return

File: refimpl/Pizza.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System;
using System.Linq;
using System.Text.RegularExpressions;

namespace Challenges
{
    public class Pizza
    {
        public string Execute(int cost)
        {
            int value;

            if (cost < 2000)
            {
                value = ((cost + 99) / 100) * 100;
            }
            else if (cost < 5000)
            {
                value = ((cost + 499) / 500) * 500;
            }
            else
            {
                var tmp = (int)(cost * 1.1);
                var rem = tmp % 1000;
                if (rem < 250) tmp -= rem;
                else if (rem > 750) tmp += (1000 - rem);
                else tmp += (500 - rem);

                value = tmp;
            }

            var tip = value - cost;

            return tip == 0 || tip > 1500
                    ? "PAID BY CARD"
                    : tip % 100 > 0
                        ? "KEEP THE CHANGE YOU FILTY ANIMAL"
                        : "HERE IS " + value;
        }
    }
}
